rfcs/0086-create-secret.md

```diff
@@ -0,0 +1,59 @@
+---
+feature: New Syntax Create Secret
+authors:
+  - "TabVersion"
+start_date: "2024/03/18"
+---
+
+# Introducing SECRETs for Secure Storage of Sensitive Information
+
+## Background
+
+This RFC proposes a new syntax and implementation for securely storing and referencing sensitive information,
+such as database passwords and API keys, in SQL statements using a SECRET catalog. 
+The goal is to enhance the security of production environments by encrypting sensitive data and applying access controls.
+
+## Proposed Syntax
+
+```sql
+-- create secret
+create scecret <secret-name> as <expr>
+```
+
+We always store SECRETs as bytea type, which maximally ensures that the information we store can adapt to different usage requirements.
+
+```sql
+-- use secret
+create table t ( ... )
+with (
+    ...
+    properties.sasl.password = secret <secret-name>
```

Comment:
```suggestion
    properties.sasl.password = secret <secret-name>
```

Don't we support function call here? Or otherwise I think a call is more straightforward.

Reply:
Actually it is a part inside with clause and ref the secret name directly will make the parser simpler.